compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/FakeInliningLocalVariablesLowering.kt

```diff
@@ -49,10 +48,14 @@ internal class FakeInliningLocalVariablesLowering(val context: JvmBackendContext
         // Do not record inline function callable reference parameters. They will not be used.
     }
 
-    override fun handleInlineFunctionLambdaParam(lambda: IrFunction, callee: IrFunction, callSite: IrDeclaration?) {
+    override fun handleInlineFunctionLambdaParam(lambdaReference: IrFunctionReference, callee: IrFunction, callSite: IrDeclaration?) {
         // Do not record lambda parameters. Instead deal with them now.
+        val lambda = lambdaReference.symbol.owner
         if (lambda.origin == IrDeclarationOrigin.LOCAL_FUNCTION_FOR_LAMBDA) {
-            lambda.addFakeInliningLocalVariablesForArguments(callee)
+            val argumentToFunctionName = context.methodSignatureMapper.mapFunctionName(callee)
```

Comment:
Unrelated to your changes, but it's a bit sad that we have dependencies of IR lowerings on `MethodSignatureMapper`, which I initially hoped to be codegen-only logic (and maybe even enforcing it by separating lowerings and codegen into different modules). The reason why I wanted that was to avoid hacks in `MethodSignatureMapper` to support the state of IR at different stages between lowerings. But looks like there's not a lot of hacks anyway there, so never mind. :)